### DISCUSSION TITLE
apple2/crt0.s imports undefined __ONCE_LOAD__ & __ONCE_SIZE__

### DISCUSSION THREAD

**Discussion by gungwald** [cc65/cc65#1901] [Q&A]
I can't find definitions for these constants anywhere in the source code, yet apple2/crt0.s tries to import them. I would submit a pull request if I knew the correct way to fix it, but I don't. I'm attaching Bill Buckels example from here: http://www.appleoldies.ca/cc65/programs/dhgr/dhrbounce.zip on this page: http://www.appleoldies.ca/bmp2dhr/sprites/. This example illustrates the failure:
> ```
> bill@panthera-spelaea:~/Downloads/dhrbounce$ make
> cl65 -O -t apple2enh -C bounce.cfg bounce.c
> ld65: Warning: bounce.cfg:44: Segment 'ZPSAVE' does not exist
> apple2/crt0.s:135: Error: Unresolved external '__ONCE_LOAD__'
> apple2/crt0.s:136: Error: Unresolved external '__ONCE_LOAD__'
> apple2/crt0.s:142: Error: Unresolved external '__ONCE_LOAD__'
> apple2/crt0.s:143: Error: Unresolved external '__ONCE_LOAD__'
> apple2/crt0.s:135: Error: Unresolved external '__ONCE_SIZE__'
> apple2/crt0.s:136: Error: Unresolved external '__ONCE_SIZE__'
> apple2/crt0.s:142: Error: Unresolved external '__ONCE_SIZE__'
> apple2/crt0.s:143: Error: Unresolved external '__ONCE_SIZE__'
> bounce.cfg:8: Error: Unresolved external '__ZPSAVE_RUN__'
> ld65: Error: 3 unresolved external(s) found - cannot create output file
> make: *** [Makefile:4: bounce.PRG] Error 1
> ```

**Comment by mrdudz** [comment]
The problem is that the example contains a .cfg file - the format of the linker configs has changed since that program was created (almost 10 years ago!). It compiles with just
```
cl65 -O -t apple2enh -o BOUNCE bounce.c
```
I cant tell if you'll need a custom linker config for this program - so test if it works or not :) If not, have a look at the apple2enh.cfg in the current cc65 tree, copy it to the program directory and modify it as needed.

**Comment by jsymolon** [comment]
What I ended up doing, is removing the "-C" config file option and use the "-t" as I had the same issue.

ld65 -v -t apple2enh -L /opt/cc65/lib -o hello text.o hello.o apple2enh.lib
